packages/expo-location/src/ExpoLocation.web.ts

```diff
@@ -49,30 +49,33 @@ function isLocationValid(location: LocationObject, options: LocationLastKnownOpt
 }
 
 /**
- * Gets the permission details. The implementation is not very good as it actually requests
- * for the current location, but there is no better way on web so far :(
+ * Gets the permission details. The implementation is not very good as it's not
+ * possible to query for permission on all browsers, apparently only the
+ * latest versions will support this.
  */
 async function getPermissionsAsync(): Promise<PermissionResponse> {
```

Comment:
```suggestion
async function getPermissionsAsync(): Promise<PermissionResponse> {
  if (!navigator?.permissions?.query) {
    throw new UnavailabilityError(
      'expo-location',
      'navigator.permissions API is not available'
    );
  }
```
Let's check if the API is available.